experimental/PropertyDDS/packages/property-inspector-table/src/NewDataForm.tsx

```diff
@@ -166,48 +166,35 @@ export const NewDataForm: React.FunctionComponent<INewDataFormProps> = (props) =
 
 	// Reshape the 'options' array which into an object suitable for consumption by react-select.
 	// Also into each option add an SVG icon corresponding to its label.
-	const typeOptions: DecoratedSelectGroupedOptionsType = options.map((group) => ({
+	const allTypeOptions: DecoratedSelectGroupedOptionsType = options.map((group) => ({
 		label: group[0],
 		options: addCorrespondingSvgIcon(group[1]),
 	}));
+	let typeOptions = allTypeOptions;
 
 	let listOfContextOptions: DecoratedSelectOptionsType = contextOptions;
-	let defaultTypeOption: IDecoratedSelectOptionType;
+	let defaultTypeOption: IDecoratedSelectOptionType = typeOptions[0].options[0];
 	const defaultContainerOption: IDecoratedSelectOptionType = listOfContextOptions[0];
 
-	const excludeUninheritedTemplates = () => {
-		typeOptions.forEach((subType) => {
-			const subTypeOptions: IDecoratedSelectOptionType[] = [];
-			subType.options.forEach((typ) => {
-				const parentTypes = PropertyFactory.getAllParentsForTemplate(typ.value);
-				if (
-					typ.value === rowData.parent!.getTypeid() ||
-					parentTypes.includes(rowData.parent!.getTypeid())
-				) {
-					subTypeOptions.push(typ);
-				}
-			});
-			subType.options = subTypeOptions;
```

Comment:
Changes in this file are caused by this line since original `Group` interface from `react-select` declares `options` as readonly. In order to avoid extending `react-select` types and increase the complexity in type domain, I decided to re-write the code here in a more "pure-functional" way and avoid inline array changes and attribute overwrites.